### DISCUSSION TITLE
Plugin NE2K not loaded

### DISCUSSION THREAD

**Discussion by stlintel** [bochs-emu/Bochs#242] [Q&A] [answered]
Usually I don't do networking with Bochs but recently had to.
When I start Bochs with Ne2K configured I get immediate PANIC

**00000000000p[DEV   ] >>PANIC<< Plugin 'ne2k' at PCI slot #2 not loaded**

```
00000000000i[PLUGIN] init_dev of 'ne2k' plugin device by virtual method
00000000000i[NE2K0 ] NE2000 NIC #0 initialized port 0x300/32 irq 3 mac b0:c4:20:00:00:00
00000000000i[NE2K0 ] win32 network driver
00000000000p[DEV   ] >>PANIC<< Plugin 'ne2k' at PCI slot #2 not loaded
...
00000000000i[PLUGIN] register state of 'ne2k' plugin device by virtual method
00000000000i[PLUGIN] reset of 'ne2k' plugin device by virtual method
```
But if I press continue Bochs runs something, don't know if NE2K is active.

[bochsrc.win10-test.txt](https://github.com/bochs-emu/Bochs/files/14071559/bochsrc.win10-test.txt)

My bochsrc attached

**Comment by vruppert** [comment, marked as answer]
At first sight I thought that this config should work. Then I had a look at the ne2k config parser. This config would work if the "pci" line appears before the "ne2k" one. For the first ne2k card it checks whether or not the ne2k is already assigned to a PCI slot (compatibility with older Bochs versions). For additional cards specifying the type is mandatory. Instead of moving the "pci" line you could also add "type=pci" in the ne2k line.

**Reply by stlintel** [reply]
PANIC is very severe and also very confusing ... Can we wait with such PANICs until everything is loaded and understood ?
I wonder why it can't work out-of-the-box regardless of the order of entries in the config.

**Reply by vruppert** [reply]
I agree, but it is tricky for this combined PCI/ISA device. The parser needs to check the config as much as it can, but the main coherency check needs to be done in the device init code. I'll have a look what I can do.

**Reply by vruppert** [reply]
Update: I have now modified the ne2k code to check for PCI device in the init code instead of the parser code. We should have a look at other devices to make sure we don't handle config parameter dependencies in the parser code.
